src/rpc/net.cpp

```diff
@@ -966,6 +966,60 @@ static RPCHelpMan addpeeraddress()
     };
 }
 
+static RPCHelpMan getaddrmaninfo()
+{
+    return RPCHelpMan{"getaddrmaninfo",
+                      "\nReturns count of addresses in new/tried table of all networks (or a given network). This RPC is for testing only.\n",
+                      {
+                              {"network", RPCArg::Type::STR, RPCArg::Optional::OMITTED, "Network type (" + Join(GetNetworkNames(), ", ") + ")"},
+                      },
+                      RPCResult{
+                              RPCResult::Type::ARR, "", "",
+                              {
+                                      {RPCResult::Type::OBJ, "", "",
+                                       {
+                                               {RPCResult::Type::STR, "network",  "The network (" + Join(GetNetworkNames(), ", ") + ")"},
+                                               {RPCResult::Type::NUM, "new", "number of addresses in new table"},
+                                               {RPCResult::Type::NUM, "tried", "number of addresses in tried table"},
+                                       }},
+                              }
+                      },
+                      RPCExamples{
+                              HelpExampleCli("getaddrmaninfo", "\"ipv4\"")
+                              + HelpExampleRpc("getaddrmaninfo", "\"ipv4\"")
+                      },
+                      [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+                      {
+                          NodeContext& node = EnsureAnyNodeContext(request.context);
+                          if (!node.addrman) {
+                              throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Address manager functionality missing or disabled");
+                          }
+
+                          const std::optional<Network> network{request.params[0].isNull() ? std::nullopt : std::optional<Network>{ParseNetwork(request.params[0].get_str())}};
+                          if (network == NET_UNROUTABLE) {
+                              throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Network not recognized: %s", request.params[0].get_str()));
```

Comment:
You could cover this error in the functional test

Reply:
thanks! done.